README.md

```diff
@@ -71,6 +71,66 @@ of [pretrained embeddings](https://finalfusion.github.io/pretrained) is offered
 and we have converted the fastText Wikipedia and Common Crawl embeddings to `finalfusion`.
 More information can also be found at https://finalfusion.github.io.
 
+## Which type of storage should I use?
+
+### Quantized embeddings
+
+Quantized embeddings store embeddings as discrete
+representations. Imagine that for a given embeddings space, you would
+find 256 prototypical embeddings. Each embedding could then be stored
+as a 1-byte pointer to one of these prototypical embeddings. Of
+course, having only 256 possible representations, this quantized
+embedding space would be very coarse-grained.
+
+**product quantizers** (**pq**) solve this problem by splitting each
+embedding evenly into *q* subvectors and finding prototypical vectors
+for each set of subvectors. If we use 256 prototypical representations
+for each subspace, *256^q* different word embeddings can be
+represented. For instance, if *q = 150*, we could represent *250^150*
+different embeddings. Each embedding would then be stored as 150
+byte-sized pointers.
+
+**optimized product quantizers** (**opq**) additionally applies a
+linear map to the embedding space to distribute variance across
+embedding dimensions.
+
+By quantizing an embedding matrix, its size can be reduced both on
+disk and in memory.
+
+### Memory mapped embeddings
+
+Normally, we read embeddings into memory. However, as an alternative
+the embeddings can be **memory mapped**. [Memory
+mapping](https://en.wikipedia.org/wiki/Memory-mapped_file) makes the
+on-disk embedding matrix available as pages in virtual memory. The
+operating system will then (transparently) load these pages into
+physical memory as necessary.
+
+Memory mapping speeds up the initial loading time of word embeddings,
+since only the vocabulary needs to be read. The operating system will
+then load (part of the) embedding matrix a by-need basis. The
+operating system can additionally free up the memory again when no
+embeddings are looked up and other processes require memory.
+
+### Empirical comparison
+
+The following empirical comparison of embedding types uses an
+embedding matrix with 2,807,440 embeddings (710,288 word, 2,097,152
+subword) of dimensionality 300. The embedding lookup timings were done
+on an Intel Core i5-8259U CPU, 2.30GHz.
+
+*Known lookup* and *Unknown lookup* time lookups of words that are
+inside/outside the vocabulary. *Lookup* contains a mixture of known
+and unknown words.
+
+| Storage    | Lookup | Known lookup | Unknown lookup |   Memory |     Disk |
+|:-----------|-------:|-------------:|---------------:|---------:|---------:|
+| array      | 449 ns |       232 ns |          18 μs | 3213 MiB | 3213 MiB |
+| array mmap | 833 ns |       494 ns |          23 μs | Variable | 3213 MiB |
+| opq        |  40 μs |        21 μs |         962 μs |  402 MiB |  402 MiB |
+| opq mmap   |  41 μs |        21 μs |         960 μs | Variable |  402 MiB |
+
```

Comment:
µ and n are similar enough that I just spent a good minute wondering how on earth `opq` could be 10 times faster than `array`.

Reply:
Could that be a font issue? For me they look really different.

![Screenshot from 2019-10-01 13-46-22](https://user-images.githubusercontent.com/49398/65959353-0989fc80-e452-11e9-97b6-af2ed63ad8b2.png)

Reply:
Rather a "not-paying-attention" issue. I was focusing on the numbers, apparently my mind didn't expect to encounter different units.

Reply:
![image](https://user-images.githubusercontent.com/30175181/65959560-9e8cf580-e452-11e9-8460-379f6bc8e19d.png)

Reply:
Maybe we could add speedups compared to the best number, let me try how that looks.

Reply:
Too busy, I have now added a note to warn for unit differences.